File: algorithms/0476.number_complement/number_complement.rs
impl Solution {
    pub fn find_complement(num: i32) -> i32 {
        let mut bits = 1i32 << 30;
        while bits & num == 0 {
            bits >>= 1;
        }
        bits <<= 1;
        bits - num - 1
    }
}

File: algorithms/0476.number_complement/number_complement.cpp
class Solution {
public:
    int findComplement(int num) {
        int s = -1;
        for (int i = 31; i >= 0; --i) {
            if (((num >> i) & 0x1) != 0) {
                s = i;
                break;
            }
        }
        int result = 0;
        for (int i = 0; i < s; ++i) {
            if (((num >> i) & 0x1) == 0) {
                result |= (0x1 << i);
            }
        }
        return result;
    }
};